Segmentation fault error in tests

This is related to #62.
We found out using `parallel=True` in `xarray.open_mfdataset` in [`fapar_lai.py`](https://github.com/EcoExtreML/zampy/blob/4f652d6ee4bdfdf91bfc0a9adffcc3b7e9229392/src/zampy/datasets/fapar_lai.py#L154) returns segmentation fault errors on linux and macos, see GA action [ubuntu](https://github.com/EcoExtreML/zampy/actions/runs/11859581857/job/33052860101?pr=62) and [mac](https://github.com/EcoExtreML/zampy/actions/runs/11859581857/job/33052861034?pr=62).

<details><summary>Example log of GA action on linux
</summary>

```
Run hatch run fast-test
Creating environment: default
Installing project in development mode
Checking dependencies
============================= test session starts ==============================
platform linux -- Python 3.11.10, pytest-[8](https://github.com/EcoExtreML/zampy/actions/runs/11859581857/job/33052860101?pr=62#step:6:9).3.3, pluggy-1.5.0
rootdir: /home/runner/work/zampy/zampy
configfile: pyproject.toml
testpaths: tests
plugins: anyio-4.6.2.post1, cov-6.0.0, mock-3.14.0
collected 85 items / 6 deselected / 7[9](https://github.com/EcoExtreML/zampy/actions/runs/11859581857/job/33052860101?pr=62#step:6:10) selected

tests/test_cds_utils.py ..............                                   [ 17%]
tests/test_converter.py ......                                           [ 25%]
tests/test_dataset_protocol.py ......                                    [ 32%]
tests/test_datasets/test_cams.py ....                                    [ 37%]
tests/test_datasets/test_era5.py ....                                    [ 43%]
Fatal Python error: Segmentation fault

Current thread 0x00007f947e3fe640 (most recent call first):
  Garbage-collecting
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/xarray/backends/netCDF4_.py", line 83 in __setitem__
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/dask/array/core.py", line 4557 in load_store_chunk
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/dask/array/core.py", line 4575 in store_chunk
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/dask/_task_spec.py", line 651 in __call__
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/dask/local.py", line 229 in execute_task
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/dask/local.py", line 243 in <listcomp>
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/dask/local.py", line 243 in batch_execute_tasks
  File "/opt/hostedtoolcache/Python/3.11.[10](https://github.com/EcoExtreML/zampy/actions/runs/11859581857/job/33052860101?pr=62#step:6:11)/x64/lib/python3.11/concurrent/futures/thread.py", line 58 in run
  File "/opt/hostedtoolcache/Python/3.[11](https://github.com/EcoExtreML/zampy/actions/runs/11859581857/job/33052860101?pr=62#step:6:12).10/x64/lib/python3.11/concurrent/futures/thread.py", line 83 in _worker
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/threading.py", line 982 in run
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/threading.py", line 1045 in _bootstrap_inner
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/threading.py", line 1002 in _bootstrap

Thread 0x00007f947f3ff640 (most recent call first):
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/xarray/backends/locks.py", line 64 in __enter__
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/xarray/backends/locks.py", line 231 in __enter__
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/xarray/backends/netCDF4_.py", line 115 in _getitem
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/xarray/core/indexing.py", line 1018 in explicit_indexing_adapter
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/xarray/backends/netCDF4_.py", line 104 in __getitem__
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/xarray/core/indexing.py", line 657 in get_duck_array
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/xarray/core/indexing.py", line 794 in get_duck_array
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/xarray/core/indexing.py", line 583 in get_duck_array
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/xarray/core/indexing.py", line 578 in __array__
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/dask/array/core.py", line [12](https://github.com/EcoExtreML/zampy/actions/runs/11859581857/job/33052860101?pr=62#step:6:13)2 in getter
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/dask/_task_spec.py", line 651 in __call__
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/dask/local.py", line 229 in execute_task
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/dask/local.py", line 243 in <listcomp>
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/dask/local.py", line 243 in batch_execute_tasks
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/concurrent/futures/thread.py", line 58 in run
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/concurrent/futures/thread.py", line 83 in _worker
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/threading.py", line 982 in run
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/threading.py", line 1045 in _bootstrap_inner
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/threading.py", line 1002 in _bootstrap

Thread 0x00007f9486921640 (most recent call first):
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/threading.py", line 331 in wait
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/threading.py", line 629 in wait
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/tqdm/_monitor.py", line 60 in run
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/threading.py", line 1045 in _bootstrap_inner
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/threading.py", line 1002 in _bootstrap

Thread 0x00007f9487922640 (most recent call first):
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/threading.py", line 331 in wait
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/threading.py", line 629 in wait
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/tqdm/_monitor.py", line 60 in run
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/threading.py", line 1045 in _bootstrap_inner
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/threading.py", line 1002 in _bootstrap

Thread 0x00007f94adc7cb80 (most recent call first):
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/threading.py", line 327 in wait
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/queue.py", line 171 in get
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/dask/local.py", line 140 in queue_get
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/dask/local.py", line 505 in get_async
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/dask/threaded.py", line 91 in get
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/dask/base.py", line 397 in compute_as_if_collection
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/dask/array/core.py", line 1282 in store
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/xarray/namedarray/daskmanager.py", line 249 in store
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/xarray/backends/common.py", line 277 in sync
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/xarray/backends/api.py", line 1881 in to_netcdf
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/xarray/core/dataset.py", line 2346 in to_netcdf
  File "/home/runner/work/zampy/zampy/src/zampy/datasets/cds_utils.py", line 394 in convert_to_zampy
  File "/home/runner/work/zampy/zampy/src/zampy/datasets/ecmwf_dataset.py", line 99 in ingest
  File "/home/runner/work/zampy/zampy/tests/test_datasets/test_era5_land.py", line 90 in ingest_dummy_data
  File "/home/runner/work/zampy/zampy/tests/test_datasets/test_era5_land.py", line [13](https://github.com/EcoExtreML/zampy/actions/runs/11859581857/job/33052860101?pr=62#step:6:14)5 in test_convert
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/_pytest/python.py", line [15](https://github.com/EcoExtreML/zampy/actions/runs/11859581857/job/33052860101?pr=62#step:6:16)9 in pytest_pyfunc_call
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/pluggy/_callers.py", line 103 in _multicall
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/pluggy/_hooks.py", line 513 in __call__
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/_pytest/python.py", line [16](https://github.com/EcoExtreML/zampy/actions/runs/11859581857/job/33052860101?pr=62#step:6:17)27 in runtest
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/_pytest/runner.py", line [17](https://github.com/EcoExtreML/zampy/actions/runs/11859581857/job/33052860101?pr=62#step:6:18)4 in pytest_runtest_call
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/pluggy/_callers.py", line 103 in _multicall
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/pluggy/_hooks.py", line 513 in __call__
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/_pytest/runner.py", line 242 in <lambda>
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/_pytest/runner.py", line 341 in from_call
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/_pytest/runner.py", line 241 in call_and_report
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/_pytest/runner.py", line 132 in runtestprotocol
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/_pytest/runner.py", line 113 in pytest_runtest_protocol
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/pluggy/_callers.py", line 103 in _multicall
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/pluggy/_hooks.py", line 513 in __call__
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/_pytest/main.py", line 362 in pytest_runtestloop
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/pluggy/_callers.py", line 103 in _multicall
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/pluggy/_hooks.py", line 513 in __call__
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/_pytest/main.py", line 337 in _main
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/_pytest/main.py", line 283 in wrap_session
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/_pytest/main.py", line 330 in pytest_cmdline_main
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/pluggy/_callers.py", line 103 in _multicall
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/pluggy/_hooks.py", line 513 in __call__
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/_pytest/config/__init__.py", line 175 in main
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/lib/python3.11/site-packages/_pytest/config/__init__.py", line 201 in console_main
  File "/home/runner/.local/share/hatch/env/virtual/zampy/_HbVvQu8/zampy/bin/pytest", line 8 in <module>

Extension modules: numpy._core._multiarray_umath, numpy.linalg._umath_linalg, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._mt[19](https://github.com/EcoExtreML/zampy/actions/runs/11859581857/job/33052860101?pr=62#step:6:20)937, numpy.random.mtrand, numpy.random._philox, numpy.random._pcg64, numpy.random._sfc64, numpy.random._generator, pandas._libs.tslibs.ccalendar, pandas._libs.tslibs.np_datetime, pandas._libs.tslibs.dtypes, pandas._libs.tslibs.base, pandas._libs.tslibs.nattype, pandas._libs.tslibs.timezones, pandas._libs.tslibs.fields, pandas._libs.tslibs.timedeltas, pandas._libs.tslibs.tzconversion, pandas._libs.tslibs.timestamps, pandas._libs.properties, pandas._libs.tslibs.offsets, pandas._libs.tslibs.strptime, pandas._libs.tslibs.parsing, pandas._libs.tslibs.conversion, pandas._libs.tslibs.period, pandas._libs.tslibs.vectorized, pandas._libs.ops_dispatch, pandas._libs.missing, pandas._libs.hashtable, pandas._libs.algos, pandas._libs.interval, pandas._libs.lib, pandas._libs.ops, pandas._libs.hashing, pandas._libs.arrays, pandas._libs.tslib, pandas._libs.sparse, pandas._libs.internals, pandas._libs.indexing, pandas._libs.index, pandas._libs.writers, pandas._libs.join, pandas._libs.window.aggregations, pandas._libs.window.indexers, pandas._libs.reshape, pandas._libs.groupby, pandas._libs.json, pandas._libs.parsers, pandas._libs.testing, cftime._cftime, yaml._yaml, psutil._psutil_linux, psutil._psutil_posix, markupsafe._speedups, scipy._lib._ccallback_c, charset_normalizer.md, scipy.sparse._sparsetools, _csparsetools, scipy.sparse._csparsetools, scipy.linalg._fblas, scipy.linalg._flapack, scipy.linalg.cython_lapack, scipy.linalg._cythonized_array_utils, scipy.linalg._solve_toeplitz, scipy.linalg._decomp_lu_cython, scipy.linalg._matfuncs_sqrtm_triu, scipy.linalg.cython_blas, scipy.linalg._matfuncs_expm, scipy.linalg._decomp_update, scipy.sparse.linalg._dsolve._superlu, scipy.sparse.linalg._eigen.arpack._arpack, scipy.sparse.linalg._propack._spropack, scipy.sparse.linalg._propack._dpropack, scipy.sparse.linalg._propack._cpropack, scipy.sparse.linalg._propack._zpropack, scipy.sparse.csgraph._tools, scipy.sparse.csgraph._shortest_path, scipy.sparse.csgraph._traversal, scipy.sparse.csgraph._min_spanning_tree, scipy.sparse.csgraph._flow, scipy.sparse.csgraph._matching, scipy.sparse.csgraph._reordering, scipy._lib._uarray._uarray, scipy.special._ufuncs_cxx, scipy.special._ufuncs, scipy.special._specfun, scipy.special._comb, scipy.special._ellip_harm_2, scipy.fftpack.convolve, zstandard.backend_c, requests.packages.charset_normalizer.md, requests.packages.chardet.md, PIL._imaging, kiwisolver._cext, rasterio._err, rasterio._filepath, rasterio._version, rasterio._env, rasterio.crs, rasterio._transform, rasterio._base, rasterio._features, rasterio._warp, rasterio._io, rasterio._vsiopener, tornado.speedups, msgpack._cmsgpack, pyproj._compat, pyproj._context, pyproj._network, pyproj._version, pyproj._geod, pyproj.list, pyproj._crs, pyproj.database, pyproj._transformer, pyproj._sync, netCDF4._netCDF4 (total: 1[20](https://github.com/EcoExtreML/zampy/actions/runs/11859581857/job/33052860101?pr=62#step:6:21))
Segmentation fault (core dumped)
tests/test_datasets/test_era5_land.py ...
Error: Process completed with exit code 139.
```
</details>

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,8 +6,25 @@ and this project adheres to [Semantic Versioning](https://semver.org/).
 
 ## Unreleased
 
+## 0.3.0 (...)
+
+Zampy works with [new CDS and ADS
+API](https://confluence.ecmwf.int/display/CKB/Please+read%3A+CDS+and+ADS+migrating+to+new+infrastructure%3A+Common+Data+Store+%28CDS%29+Engine).
+
+### Added
+
+The zampy test dataset is generated using data from the new API, [PR
+  #62](https://github.com/EcoExtreML/zampy/pull/62).
+
+### Fixed
+
+- Add supports for the new CDS API [PR
+  #62](https://github.com/EcoExtreML/zampy/pull/62)
+- Fixed segmentation fault error [PR
+  #66](https://github.com/EcoExtreML/zampy/pull/66)
 
 ## 0.2.0 (2024-09-02)
+
 First release of `zampy`.
 
 `zampy` is designed to retrieve data for LSM model input. It can help you prepare the data within the following steps:

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -47,13 +47,14 @@ classifiers = [
   "Programming Language :: Python :: 3.11",
 ]
 dependencies = [
+  "h5netcdf",  # see issue 65
   "requests",
   "pyyaml",
   "netcdf4",
   "numpy",
   "pandas",
   "matplotlib",
-  "xarray[io]",
+  "xarray",
   "scipy",  # required for xarray.interpolate
   "rioxarray",  # required for TIFF files
   "tqdm",

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -53,7 +53,7 @@ dependencies = [
   "numpy",
   "pandas",
   "matplotlib",
-  "xarray",
+  "xarray[io]",
   "scipy",  # required for xarray.interpolate
   "rioxarray",  # required for TIFF files
   "tqdm",

--- a/src/zampy/datasets/cds_utils.py
+++ b/src/zampy/datasets/cds_utils.py
@@ -391,7 +391,7 @@ def convert_to_zampy(
         ds = parse_nc_file(file)
         # Rename the vswl data:
         ncfile = Path(str(ncfile).replace("volumetric_soil_water", "soil_moisture"))
-        ds.to_netcdf(path=ncfile)
+        ds.to_netcdf(path=ncfile, engine="h5netcdf")
 
 
 var_reference_ecmwf_to_zampy = {
@@ -444,7 +444,7 @@ def parse_nc_file(file: Path) -> xr.Dataset:
         CF/Zampy formatted xarray Dataset
     """
     # Open chunked: will be dask array -> file writing can be parallelized.
-    ds = xr.open_dataset(file, chunks={"x": 50, "y": 50})
+    ds = xr.open_dataset(file, chunks={"x": 50, "y": 50}, engine="h5netcdf")
 
     for variable in ds.variables:
         if variable in var_reference_ecmwf_to_zampy:

--- a/src/zampy/datasets/ecmwf_dataset.py
+++ b/src/zampy/datasets/ecmwf_dataset.py
@@ -119,7 +119,9 @@ class ECMWFDataset:  # noqa: D101
             if var in variable_names:
                 files += (ingest_dir / self.name).glob(f"{self.name}_{var}*.nc")
 
-        ds = xr.open_mfdataset(files, chunks={"latitude": 200, "longitude": 200})
+        ds = xr.open_mfdataset(
+            files, chunks={"latitude": 200, "longitude": 200}, engine="h5netcdf"
+        )
 
         # rename valid_time to time
         if "valid_time" in ds.dims:
@@ -152,7 +154,7 @@ class ECMWFDataset:  # noqa: D101
         for file in data_files:
             # start conversion process
             print(f"Start processing file `{file.name}`.")
-            ds = xr.open_dataset(file, chunks={"x": 50, "y": 50})
+            ds = xr.open_dataset(file, chunks={"x": 50, "y": 50}, engine="h5netcdf")
             ds = converter.convert(ds, dataset=self, convention=convention)
             # TODO: support derived variables
             # TODO: other calculations

--- a/src/zampy/datasets/eth_canopy_height.py
+++ b/src/zampy/datasets/eth_canopy_height.py
@@ -135,7 +135,9 @@ class EthCanopyHeight:  # noqa: D101
         if self.variable_names[1] in variable_names:
             files += (ingest_dir / self.name).glob("*Map_SD.nc")
 
-        ds = xr.open_mfdataset(files, chunks={"latitude": 2000, "longitude": 2000})
+        ds = xr.open_mfdataset(
+            files, chunks={"latitude": 2000, "longitude": 2000}, engine="h5netcdf"
+        )
         ds = ds.sel(time=slice(time_bounds.start, time_bounds.end))
 
         grid = xarray_regrid.create_regridding_dataset(
@@ -159,7 +161,7 @@ class EthCanopyHeight:  # noqa: D101
         for file in data_files:
             # start conversion process
             print(f"Start processing file `{file.name}`.")
-            ds = xr.open_dataset(file, chunks={"x": 2000, "y": 2000})
+            ds = xr.open_dataset(file, chunks={"x": 2000, "y": 2000}, engine="h5netcdf")
             ds = converter.convert(ds, dataset=self, convention=convention)
             # TODO: support derived variables
             # TODO: other calculations
@@ -249,10 +251,7 @@ def convert_tiff_to_netcdf(
         ds = ds.interpolate_na(dim="longitude", limit=1)
         ds = ds.interpolate_na(dim="latitude", limit=1)
 
-        ds.to_netcdf(
-            path=ncfile,
-            encoding=ds.encoding,
-        )
+        ds.to_netcdf(path=ncfile, encoding=ds.encoding, engine="h5netcdf")
 
 
 def parse_tiff_file(file: Path, sd_file: bool = False) -> xr.Dataset:

--- a/src/zampy/datasets/fapar_lai.py
+++ b/src/zampy/datasets/fapar_lai.py
@@ -150,7 +150,7 @@ class FaparLAI:  # noqa: D101
         variable_names: list[str],
     ) -> xr.Dataset:
         files = list((ingest_dir / self.name).glob("*.nc"))
-        ds = xr.open_mfdataset(files)  # see issue 65
+        ds = xr.open_mfdataset(files, engine="h5netcdf")  # see issue 65
         ds = ds.sel(time=slice(time_bounds.start, time_bounds.end))
 
         grid = xarray_regrid.create_regridding_dataset(
@@ -175,7 +175,9 @@ class FaparLAI:  # noqa: D101
         for file in data_files:
             # start conversion process
             print(f"Start processing file `{file.name}`.")
-            ds = xr.open_dataset(file, chunks={"latitude": 2000, "longitude": 2000})
+            ds = xr.open_dataset(
+                file, chunks={"latitude": 2000, "longitude": 2000}, engine="h5netcdf"
+            )
             ds = converter.convert(ds, dataset=self, convention=convention)
 
         return True
@@ -248,7 +250,9 @@ def download_fapar_lai(
 def ingest_ncfile(ncfile: Path, ingest_folder: Path) -> None:
     """Ingest the 'raw' netCDF file to the Zampy standard format."""
     print(f"Converting file {ncfile.name}...")
-    ds = xr.open_dataset(ncfile, decode_times=False, chunks={"lat": 5000, "lon": 5000})
+    ds = xr.open_dataset(
+        ncfile, decode_times=False, chunks={"lat": 5000, "lon": 5000}, engine="h5netcdf"
+    )
     ds = ds.rename(
         {
             "LAI": "leaf_area_index",
@@ -260,6 +264,7 @@ def ingest_ncfile(ncfile: Path, ingest_folder: Path) -> None:
     ds[["leaf_area_index"]].to_netcdf(
         path=ingest_folder / ncfile.name,
         encoding={"leaf_area_index": {"zlib": True, "complevel": 3}},
+        engine="h5netcdf",
     )
     ds.close()  # explicitly close to release file to system (for Windows)
 

--- a/src/zampy/datasets/land_cover.py
+++ b/src/zampy/datasets/land_cover.py
@@ -136,7 +136,9 @@ class LandCover:
                 )
                 raise ValueError(msg)
         files = list((ingest_dir / self.name).glob(f"{self.name}_*.nc"))
-        ds = xr.open_mfdataset(files, chunks={"latitude": 200, "longitude": 200})
+        ds = xr.open_mfdataset(
+            files, chunks={"latitude": 200, "longitude": 200}, engine="h5netcdf"
+        )
         ds = ds.sel(time=slice(time_bounds.start, time_bounds.end))
 
         grid = xarray_regrid.create_regridding_dataset(
@@ -173,7 +175,7 @@ class LandCover:
 
         for file in data_files:
             print(f"Start processing file `{file.name}`.")
-            ds = xr.open_dataset(file)
+            ds = xr.open_dataset(file, engine="h5netcdf")
             ds = converter.convert(ds, dataset=self, convention=convention)
 
         return True
@@ -197,7 +199,7 @@ def unzip_raw_to_netcdf(
         print(f"File '{ncfile.name}' already exists, skipping...")
     else:
         ds = extract_netcdf_to_zampy(file)
-        ds.to_netcdf(path=ncfile)
+        ds.to_netcdf(path=ncfile, engine="h5netcdf")
 
 
 def extract_netcdf_to_zampy(file: Path) -> xr.Dataset:
@@ -220,7 +222,7 @@ def extract_netcdf_to_zampy(file: Path) -> xr.Dataset:
             zip_object.extract(zipped_file_name, path=unzip_folder)
 
         # only keep land cover class variable
-        with xr.open_dataset(unzip_folder / zipped_file_name) as ds:
+        with xr.open_dataset(unzip_folder / zipped_file_name, engine="h5netcdf") as ds:
             var_list = list(ds.data_vars)
             raw_variable = "lccs_class"
             var_list.remove(raw_variable)

--- a/src/zampy/datasets/prism_dem.py
+++ b/src/zampy/datasets/prism_dem.py
@@ -144,7 +144,7 @@ class PrismDEM:
             """Remove overlapping coordinates on the edges."""
             return ds.isel(latitude=slice(None, -1), longitude=slice(None, -1))
 
-        ds = xr.open_mfdataset(files, preprocess=preproc)
+        ds = xr.open_mfdataset(files, preprocess=preproc, engine="h5netcdf")
 
         grid = xarray_regrid.create_regridding_dataset(
             utils.make_grid(spatial_bounds, resolution)
@@ -168,7 +168,7 @@ class PrismDEM:
         for file in data_files:
             # start conversion process
             print(f"Start processing file `{file.name}`.")
-            ds = xr.open_dataset(file)
+            ds = xr.open_dataset(file, engine="h5netcdf")
             ds = converter.convert(ds, dataset=self, convention=convention)
 
         return True
@@ -216,6 +216,7 @@ def convert_raw_dem_to_netcdf(
         ds.to_netcdf(
             path=ncfile,
             encoding=ds.encoding,
+            engine="h5netcdf",
         )
 
 

--- a/src/zampy/recipe.py
+++ b/src/zampy/recipe.py
@@ -162,7 +162,9 @@ class RecipeManager:
             time_end = str(self.timebounds.end.astype("datetime64[Y]"))
             # e.g. "era5_2010-2020.nc"
             fname = f"{dataset_name.lower()}_{time_start}-{time_end}.nc"
-            ds.to_netcdf(path=self.data_dir / fname, encoding=encoding)
+            ds.to_netcdf(
+                path=self.data_dir / fname, encoding=encoding, engine="h5netcdf"
+            )
             del ds
 
         print(

--- a/tests/test_datasets/test_fapar_lai.py
+++ b/tests/test_datasets/test_fapar_lai.py
@@ -89,7 +89,9 @@ class TestFaparLAI:
         lai_dataset = FaparLAI()
         lai_dataset.ingest(download_dir=data_folder, ingest_dir=dummy_dir)
 
-        ds = xr.open_mfdataset((dummy_dir / "fapar-lai").glob("*.nc"))
+        ds = xr.open_mfdataset(
+            (dummy_dir / "fapar-lai").glob("*.nc"), engine="h5netcdf"
+        )
         assert isinstance(ds, xr.Dataset)
 
     def test_load(self, dummy_dir):

--- a/tests/test_recipes/test_simple_recipe.py
+++ b/tests/test_recipes/test_simple_recipe.py
@@ -48,7 +48,9 @@ def test_recipe(tmp_path: Path, mocker):
 
         rm.run()
 
-        ds = xr.open_mfdataset(str(tmp_path / "output" / "era5_recipe" / "*.nc"))
+        ds = xr.open_mfdataset(
+            str(tmp_path / "output" / "era5_recipe" / "*.nc"), engine="h5netcdf"
+        )
         assert all(var in ds.data_vars for var in ["Psurf", "Wind_N"])
         # Check if time frequency is correct
         assert ds.time.diff("time").min() == np.timedelta64(1, "h")
@@ -85,7 +87,9 @@ def test_recipe_with_lower_frequency(tmp_path: Path, mocker):
 
         rm.run()
 
-        ds = xr.open_mfdataset(str(tmp_path / "output" / "era5_recipe" / "*.nc"))
+        ds = xr.open_mfdataset(
+            str(tmp_path / "output" / "era5_recipe" / "*.nc"), engine="h5netcdf"
+        )
         # check the lenght of the time dimension, mean values are used
         assert len(ds.time) == 4
 
@@ -121,7 +125,9 @@ def test_recipe_with_higher_frequency(tmp_path: Path, mocker):
 
         rm.run()
 
-        ds = xr.open_mfdataset(str(tmp_path / "output" / "era5_recipe" / "*.nc"))
+        ds = xr.open_mfdataset(
+            str(tmp_path / "output" / "era5_recipe" / "*.nc"), engine="h5netcdf"
+        )
         # check the lenght of the time dimension, data is interpolated
         assert len(ds.time) == 47
 
@@ -156,7 +162,9 @@ def test_recipe_with_two_time_values(tmp_path: Path, mocker):
 
         rm.run()
 
-        ds = xr.open_mfdataset(str(tmp_path / "output" / "era5_recipe" / "*.nc"))
+        ds = xr.open_mfdataset(
+            str(tmp_path / "output" / "era5_recipe" / "*.nc"), engine="h5netcdf"
+        )
         # check the lenght of the time dimension
         assert len(ds.time) == 2
 
@@ -191,7 +199,9 @@ def test_recipe_with_one_time_values(tmp_path: Path, mocker):
 
         rm.run()
 
-        ds = xr.open_mfdataset(str(tmp_path / "output" / "era5_recipe" / "*.nc"))
+        ds = xr.open_mfdataset(
+            str(tmp_path / "output" / "era5_recipe" / "*.nc"), engine="h5netcdf"
+        )
         # check the lenght of the time dimension, should not do interpolation or
         # extrapolation in time
         assert len(ds.time) == 1

